crates/bifrost/src/providers/replicated_loglet/network.rs

```diff
@@ -58,10 +63,164 @@ impl RequestPump {
                 _ = &mut cancel => {
                     break;
                 }
-                Some(_append) = self.append_stream.next() => {
+                Some(append) = self.append_stream.next() => {
+                    self.handle_append(Arc::clone(&provider), append).await;
                 }
             }
         }
+
+        Ok(())
+    }
+
+    /// Infailable handle_append method
+    async fn handle_append<T: TransportConnect>(
+        &mut self,
+        provider: Arc<ReplicatedLogletProvider<T>>,
+        incoming: Incoming<Append>,
+    ) {
+        let task = HandleAppendTask {
+            metadata: self.metadata.clone(),
+            provider,
+        };
+
+        // we spawn immediately to avoid slowing the request pump loop
+        // in case responses will go over a slow channel
+        let _ = task_center().spawn_child(
+            TaskKind::Disposable,
+            "handle-append-request",
+            None,
+            task.run(incoming),
+        );
+    }
+}
+
+struct HandleAppendTask<T> {
+    provider: Arc<ReplicatedLogletProvider<T>>,
+    metadata: Metadata,
+}
+
+impl<T> HandleAppendTask<T>
+where
+    T: TransportConnect,
+{
+    fn get_loglet(
+        &self,
+        provider: &ReplicatedLogletProvider<T>,
+        header: &CommonRequestHeader,
+    ) -> Result<Arc<ReplicatedLoglet<T>>, SequencerStatus> {
```

Comment:
Why do we need to search the the metadata on every append? Can we try to find the loglet in active_loglets first by log_id/segment_id (the loglet_id should match the returned loglet) and only go down this search path if it's not already active?